webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -216,11 +218,29 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
       };
       this.contactId = null;
       formId = contactType.create_form;
+
+      await this.validateParentForCreateForm();
     }
 
     return formId;
   }
 
+  private async validateParentForCreateForm() {
+    if (!this.contact.parent) {
```

Comment:
So, the edge-case logic here is super jank.  

- `datasource.v1.place.getByUuid` will return `null` if no _place_ is found for the identifier (e.g. if `this.contact.parent` is a `person`).
- `contactTypesService.getTypeId` will return `undefined` when you give it a nullish `parent`
- `contactTypesService.getChildren` will return the "orphan" (aka "top level") types (e.g. `district_hospital`) when you give it a nullish `parentType`.  

The combination of these means that with your current logic it will still let me add a `district_hospital` as the _child_ of a `person`.

---

Additionally, another edge case that just occurred to me that is not covered is that we should throw an error if someone tries to add a contact _to the top level_ when the type of that contact is not an "orphan" (aka when there are parents configured for the contact type).  So, the user should be able to fill out a form at `contacts/add/district_hospital` since there are no parents for that type, but they should not be able to fill out `contacts/add/person` since `person` is configured as a child type.

Reply:
Thanks @jkuester . I wanted to get your review early so that we don't skip a day. I've made the checks very explicit.